### PR TITLE
Add support for old forks.

### PR DESCRIPTION
**Pull request by rakita**
Add support for all ethereum forks. From homestead onwards.

Constantinople is not added as it was superseded with Petersburg:
https://github.com/ethereum/execution-specs

All legacy eth tests are passing and CI is updated to run them.

### CODE REVIEW

File: crates/revm/src/db.rs [hunk 1]
```diff
@@ -17,14 +17,14 @@ use auto_impl::auto_impl;
 #[auto_impl(& mut, Box)]
 pub trait Database {
     /// Get basic account information.
-    fn basic(&mut self, address: H160) -> AccountInfo;
+    fn basic(&mut self, address: H160) -> Result<Option<AccountInfo>, &'static str>;
```
Comment (mattsse) [inline crates/revm/src/db.rs:20]: would it be possible to make this a `Cow`, this way we could include additional context in the error like address etc

Comment (mattsse) [inline crates/revm/src/db.rs:20]: also I think it'd be useful to use an error alias and reexport it

something like:
```rust
pub type DbResult<T> = Result<T, &'static str>;

```

Comment (rakita) [reply crates/revm/src/db.rs:20]: Adding associate type Error to database trait would maybe make this better

Comment (rakita) [reply crates/revm/src/db.rs]: You can now specify `Error` type inside `Database`.

Comment (rakita) [reply crates/revm/src/db.rs]: `Error` type fixes this

**Comment by gakonst** [comment]
legend

**Pull request merged**